pkg/types/baremetal/validation/platform_test.go

```diff
@@ -0,0 +1,195 @@
+package validation
+
+import (
+	"net"
+	"testing"
+
+	"github.com/openshift/installer/pkg/ipnet"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidatePlatform(t *testing.T) {
+	iface, _ := net.Interfaces()
+	network := &types.Networking{MachineCIDR: ipnet.MustParseCIDR("192.168.111.0/24")}
+	cases := []struct {
+		name     string
+		platform *baremetal.Platform
+		network  *types.Networking
+		expected string
+	}{
+		{
+			name: "valid",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          iface[0].Name,
+				ProvisioningBridge:      iface[0].Name,
+			},
+			network:  network,
+			expected: "",
```

Comment:
Nit: this is Go's zero value, so you can drop this line without changing anything.

Reply:
Yeah, I'm fine either way, it just seemed more explicit to do it like this, given the later conditional on an empty string for the "pass" path.

Reply:
Ok removed.